molecule/molecule-runtime/src/commonTest/java/app/cash/molecule/MoleculeTest.kt

```diff
@@ -202,4 +203,29 @@ class MoleculeTest {
 
     scope.cancel()
   }
+
+  enum class DisposableEffectState { NOT_LAUNCHED, LAUNCHED, DISPOSED }
+
+  @Test fun disposableEffectDisposesWhenScopeIsCancelled() {
+    val dispatcher = TestCoroutineDispatcher()
+    val clock = BroadcastFrameClock()
+    val scope = CoroutineScope(dispatcher + clock)
+
+    var state: DisposableEffectState = DisposableEffectState.NOT_LAUNCHED
```

Comment:
I usually use an `Int` which starts at 0 and then is incremented but this is fine (and not in the meme way).